realm/realm-library/src/androidTest/java/io/realm/RealmMigrationTests.java

```diff
@@ -823,7 +822,7 @@ public void migrationException_realmListChanged() throws IOException {
             fail();
         } catch (RealmMigrationNeededException ignored) {
             assertThat(ignored.getMessage(),
-                    CoreMatchers.containsString("Invalid RealmList type for field 'cats': 'class_Dog' expected "));
+                    CoreMatchers.containsString("Property 'CatOwner.cats' has been changed from 'array' to 'array'"));
```

Comment:
It's a bug in Object Store, https://github.com/realm/realm-object-store/pull/478 will be fixed later.